pkg/service/application_cache.go

```diff
@@ -0,0 +1,70 @@
+package service
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/pyroscope-io/pyroscope/pkg/storage"
+)
+
+type ApplicationWriter interface {
+	CreateOrUpdate(ctx context.Context, application storage.ApplicationMetadata) error
+}
+
+type ApplicationCacheService struct {
+	appSvc ApplicationWriter
+	cache  *cache
+}
+
+type ApplicationCacheServiceConfig struct {
+	Size int
+	TTL  time.Duration
+}
+
+func NewApplicationCacheService(config ApplicationCacheServiceConfig, appSvc ApplicationWriter) *ApplicationCacheService {
+	if config.Size <= 0 {
+		config.Size = 100
```

Comment:
What happens if we go over 100? I would maybe set it to something higher than that, e.g 1000.